nameable.rb

```diff
@@ -0,0 +1,31 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
+  end
+end
+
+class BaseDecorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    super().upcase
+  end
```

Comment:
- [ ] Please use the `capitalize` method instead of the `upcase` method. The `upcase` method is used to convert all characters to uppercase but for this project's requirements, we want only the first character to be uppercase.
  - You can consider using the below code.
  
```ruby
def correct_name
      @nameable.correct_name.capitalize
    end
```